src/components/structures/RoomView.js

```diff
@@ -557,7 +546,35 @@ module.exports = React.createClass({
             self.setState({
                 searchInProgress: false
             });
-        });
+        }).done();
```

Comment:
This will cause an *uncaught* exception if the HTTP request fails. Handle the failure or bubble it up by returning the Promise so something else can. I would probably `setState({ searchInProgress: false}) ` at the very least.